### PR TITLE
Disable to change group settings unless group creator tries

### LINKED ISSUE #1707: Should block updating group setting unless group creator tries
### Problem

In the current version, general group members can open modal using the URL (e.g, http://localhost:8000/app/dashboard?modal=MincomeProposal), and update group settings when the group has less than 3 members.

### Solution

- Close the modal
- Disable the submission button
Among the two options, select one and make it work in the following modals (Mincome.vue, ChangeVotingRules.vue, DistributionDate.vue).

### PR DESCRIPTION
**Pull request by Silver-IT**
### Summary of Changes

- Closes #1707

### CODE REVIEW

File: frontend/views/containers/proposals/DistributionDate.vue [hunk 1]
```diff
@@ -5,7 +5,6 @@
     :disabled='$v.form.$invalid || distributionStarted'
     :maxSteps='config.steps.length'
     :currentStep.sync='ephemeral.currentStep'
-    variant='changeDistributionDate'
```
Comment (taoeffect) [inline frontend/views/containers/proposals/DistributionDate.vue:8]: Why was this removed?

File: frontend/views/containers/proposals/ProposalTemplate.vue [hunk 2]
```diff
@@ -46,7 +46,7 @@
           :class='submitStyleNonProposal'
           v-if='!shouldPropose'
           @click='submit'
-          :disabled='disabled'
+          :disabled='disabled || !isGroupCreator'
```
Comment (taoeffect) [inline frontend/views/containers/proposals/ProposalTemplate.vue:49]: It's not clear to me why this condition is being added. Anyone should be able to create a proposal.

**Review by taoeffect — COMMENTED** [review]
Some questions.

This issue is also very low priority.

File: frontend/views/containers/proposals/DistributionDate.vue (hunk 1 above)
Comment (Silver-IT) [reply frontend/views/containers/proposals/DistributionDate.vue:8]: It's not necessary. Nowhere using it.

File: frontend/views/containers/proposals/ProposalTemplate.vue (hunk 2 above)
Comment (Silver-IT) [reply frontend/views/containers/proposals/ProposalTemplate.vue:49]:
```
button-submit(
  key='change'
  :class='submitStyleNonProposal'
  v-if='!shouldPropose'
  @click='submit'
  :disabled='disabled || !isGroupCreator'
  data-test='submitBtn'
) {{ submitTextNonProposal }}
```

As the button title says, it's submit button without making proposal. This button is visible when there are less than 3 members in the group. So if he is not group creator, he shouldn't make group settings just like `mincome`, `distributionDate`, `votingRule`, ...
`Add Member`, and `Remove Member` have different user flow so we don't need to take care of them.

**Comment by SebinSong** [comment]
@Silver-IT 
I agree that the right to make instant updates to those group-settings when the members are < 3 should only be given to the creator.
But it also occurred to me that it would be a better fix if the app somehow lets the non-creator users why they cannot make that action, instead of just disabling the submit button.

For example,
Either by showing a banner in the proposal modal saying only the group creator can make the instant update Or,
by showing a prompt pop-up(`sbp('gi.ui/prompt', ...)`) saying the same when the call-to-action is clicked.

What's your thought?

File: frontend/views/containers/proposals/ProposalTemplate.vue (hunk 2 above)
Comment (taoeffect) [reply frontend/views/containers/proposals/ProposalTemplate.vue:49]: > This button is visible when there are less than 3 members in the group

Where is this check done? `:disabled` is being set to `true` regardless of the value of `disabled` parameter, because of the `||`. So it will always be disabled if we are not the group creator.

Comment (SebinSong) [reply frontend/views/containers/proposals/ProposalTemplate.vue:49]: @taoeffect 
This `v-if` line in the above does the check there. 
```js
button-submit(
  ...
  v-if='!shouldPropose'
  ...
)
```

**Comment by Silver-IT** [comment]
> Either by showing a banner in the proposal modal saying only the group creator can make the instant update Or,
by showing a prompt pop-up(sbp('gi.ui/prompt', ...)) saying the same when the call-to-action is clicked. What's your thought?

@SebinSong. Yeah, that's good idea. However, with the normal userflow, non-creators can not access this modal. Please check `proposalOptions` function in `ProposalsWidget.vue`. What I did is to disable to change group settings when non-creator users would access modal by opening the specific URL. (Check Issue #1707) So I think we could ignore it. If most of the teammates, want to add it, I can add it for sure. However, thanks for your good opinion.

**Review by taoeffect — APPROVED** [review]
Nice - thanks for the explanations @Silver-IT! Tested & approved!
